invenio_rdm_records/services/generators.py

```diff
@@ -18,12 +19,52 @@
 from invenio_communities.generators import CommunityRoleNeed, CommunityRoles
 from invenio_communities.proxies import current_roles
 from invenio_records_permissions.generators import Generator
+from invenio_records_resources.services.files.transfer import TransferType
 from invenio_search.engine import dsl
 
 from invenio_rdm_records.records import RDMDraft
 
 
-class IfRestricted(Generator):
+class ConditionalGenerator(Generator):
+    """Generator that depends on whether a condition is true or not.
+
+    If...(
+        then_=[...],
+        else_=[...],
+    )
+    """
+
+    def __init__(self, then_, else_):
+        """Constructor."""
+        self.then_ = then_
+        self.else_ = else_
+
+    @abstractmethod
+    def _condition(self, **kwargs):
```

Comment:
I think this can be 
```suggestion
    def condition(self, **kwargs):
```

as it belongs to the public API of the class

Reply:
Discussed IRL